When compiling for android, and linking libc, you will always hit a @compileError

### Zig Version

0.12 (and above)

### Steps to Reproduce and Observed Behavior

1. Create a dummy, hello world zig file
```
// main.zig
const std = @import("std");

pub fn main() !void {
    std.debug.print("All your {s} are belong to us.\n", .{"codebase"});
}
```
2. Run `zig build-exe -target aarch64-linux-android main.zig -lc`
```
▶ zig build-exe -target aarch64-linux-android src/main.zig -lc
/opt/zig/0.12/lib/std/c.zig:1899:5: error: android bionic libc does not implement getcontext
    @compileError("android bionic libc does not implement getcontext")
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
referenced by:
    getContext: /opt/zig/0.12/lib/std/debug.zig:252:52
    writeCurrentStackTrace__anon_6682: /opt/zig/0.12/lib/std/debug.zig:785:25
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

### Expected Behavior

This should be allowed to be compiled with `getContext` being an undefined extern function. Right now, this completely prevents users from continuing. Leaving it as an extern function would allow users to be able to link their own dummy `getContext` function if they want.

--- a/lib/std/debug.zig
+++ b/lib/std/debug.zig
@@ -237,8 +237,8 @@ pub fn relocateContext(context: *ThreadContext) void {
     };
 }
 
-pub const have_getcontext = @hasDecl(posix.system, "getcontext") and
-    native_os != .openbsd and native_os != .haiku and
+pub const have_getcontext = native_os != .openbsd and native_os != .haiku and
+    !builtin.target.isAndroid() and
     (native_os != .linux or switch (builtin.cpu.arch) {
     .x86,
     .x86_64,

